src/openms/source/FORMAT/MascotRemoteQuery.cpp

```diff
@@ -701,4 +720,16 @@ namespace OpenMS
          << "<<<< Header to " << what << " (end)." << endl;
   }
 
+  Int MascotRemoteQuery::getSearchNumberFromFilePath_(const String& path)
+  {
+    Int search_number = 0;
+    int pos = path.find_last_of("/\\");
+    String tmp = path.substr(pos + 1);
+    pos = tmp.find_last_of(".");
+    tmp = tmp.substr(1, pos - 1);
+   
+    istringstream ss(tmp);
+    ss >> search_number;
+    return search_number;
```

Comment:
please use
```
return tmp.toInt();
```
since the implementation is verified to be fast on all platforms. StringStream is really bad for this.